model-optimizer/unit_tests/extensions/middle/quantize_linear_resolver_test.py

```diff
@@ -0,0 +1,483 @@
+                                 ('high_reshape_const_data', 'high_reshape'),
+                                 ('high_reshape', 'high_reshape_data'),
+                                 ('high_reshape_data', 'f_quantize'),
+
+                                 ('f_quantize', 'f_quantize_data'),
+                                 ('scale_param_q_data', 'mul2', {'out': 0}),
+                                 ('in_high', 'in_high_data'),
+                                 ('in_high_data', 'mul2'),
+                                 ('mul2', 'mul2_data'),
+                                 ('mul2_data', 'low_reshape'),
+                                 ('low_reshape', 'low_reshape_data'),
+                                 ('low_reshape_data', 'f_quantize'),
+                                 ('low_reshape_const', 'low_reshape_const_data'),
+                                 ('low_reshape_const_data', 'low_reshape'),
+                                 ('out_low', 'out_low_data'),
+                                 ('out_low_data', 'f_quantize'),
+
+                                 ('out_high', 'out_high_data'),
+                                 ('out_high_data', 'f_quantize'),
+                                 ('f_quantize_data', 'cast'),
+                                 ('cast', 'cast_data'),
+                                 ('cast_data', 'out'),
+                                 ('out', 'out_data'),
+                                 ('out_data', 'result'),
+                                 ],
+                                {'in_low': {'shape': np.array([4]), 'value': np.array([-2., -3., -4., -5.])},
+                                 'in_low_data': {'shape': np.array([4]), 'value': np.array([-2., -3., -4., -5.])},
+                                 'in_high': {'shape': np.array([4]), 'value': np.array([253., 252., 251., 250.])},
+                                 'in_high_data': {'shape': np.array([4]), 'value': np.array([253., 252., 251., 250.])},
+                                 'out_low': {'shape': np.array([]), 'value': 0},
+                                 'out_low_data': {'shape': np.array([]), 'value': 0},
+                                 'out_high': {'shape': np.array([]), 'value': 255},
+                                 'out_high_data': {'shape': np.array([]), 'value': 255},
+                                 'cast': {'dst_type': np.uint8},
+                                 'low_reshape_const_data': {'shape': np.array([4]), 'value': np.array([1, 1, 4, 1])},
+                                 'high_reshape_const_data': {'shape': np.array([4]), 'value': np.array([1, 1, 4, 1])},
+                                 }, nodes_with_edges_only=True)
+
+        graph.stage = 'middle'
+        QuantizeLinearResolver().find_and_replace_pattern(graph)
+
+        (flag, resp) = compare_graphs(graph, graph_ref, 'result', check_op_attrs=True)
+        self.assertTrue(flag, resp)
+
+    def test_quantize_uint8_with_negative(self):
```

Comment:
The only difference between the tests is that some shapes and values are different. Let's unify them using the generator. It will dramatically reduce the size of the file.